File: javadevelopers-understanding-kotlin-properties/src/test/java/com/moelholm/jav/FirstTests.java
package com.moelholm.jav;

import static org.assertj.core.api.Java6Assertions.assertThat;

import com.moelholm.First;
import org.junit.Test;

public class FirstTests {

  @Test
  public void test() {

    First first = new First();

    first.setMessageOne("Hello from Java");

    assertThat(first.getMessageOne()).isEqualTo("Hello from Java");

    assertThat(first.getMessageTwo()).isEqualTo("two");

  }

}

File: javadevelopers-understanding-kotlin-properties/src/test/kotlin/com/moelholm/kot/FirstTests.kt
package com.moelholm.kot

import com.moelholm.First
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class FirstTests {

  @Test
  fun test() {

    val first = First()

    first.messageOne = "Hello from Kotlin"

    assertThat(first.messageOne).isEqualTo("Hello from Kotlin")

    assertThat(first.messageTwo).isEqualTo("two")

  }

}